fix(integration): reorderTask doesn't reject when reference has different parent

## Context

`reorderTask(taskId, { before: referenceTaskId })` resolves `undefined` when `referenceTaskId` belongs to a different parent than `taskId`. It should reject with a `ValidationError` per the adapter contract — the contract test asserts the rejection.

**Test:** `tests/contract/adapter.contract.ts:189` — `reorderTask — ValidationError when reference has different parent`.

```text
AssertionError: promise resolved "undefined" instead of rejecting
- Expected: Error { "message": "rejected promise" }
+ Received: undefined
```

Tracks: cluster of integration test failures repro'd in [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989).

## Acceptance Criteria

- [ ] `reorderTask(a, { before: b })` rejects with `ValidationError` when `a.projectId !== b.projectId` (or when their `parentId` chains don't share a common container)
- [ ] `tests/integration/transportRouter.contract.test.ts > tasks — CRUD > reorderTask — ValidationError when reference has different parent` passes locally and on the `macos-omnifocus` runner
- [ ] Validation runs **before** the actual reorder mutation, so a failed call doesn't leave OF in a partially-reordered state
- [ ] Equivalent OmniJS path (the routing target for `reorderTask`) also enforces the validation — even if the script is the one that throws, the typed error must surface as `ValidationError` at the wrapper boundary

## Technical Notes

- `reorderTask` routes to OmniJS per [src/adapter/router.ts](src/adapter/router.ts) and the failing-test stack trace points at `OmniJsTransport.reorderTask` (line 272). The OmniJS script at `scripts/omnijs/task_reorder.js` (or wherever it lives) likely doesn't compare the parent of `taskId` and `referenceTaskId` before doing the reorder.
- Could be implemented either inside the OmniJS script (cheap; the error code path already exists) or in the Node-side wrapper before invocation (avoids OmniJS round-trip on a known-invalid input). Pick whichever matches the existing pattern in sibling validators.
- Smaller blast radius than the other tickets in this cluster — single-test failure, narrow validation gap. Good candidate to ship first.

## Dependencies

- Tracks: cluster of integration test failures repro'd in [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989)
- Independent of the `createTask projectId` cluster (different code path)

--- a/src/scripts/omnijs/task_reorder.js
+++ b/src/scripts/omnijs/task_reorder.js
@@ -55,6 +55,20 @@
         error: { code: "NOT_FOUND", message: `Reference task not found: ${refId}` },
       });
     }
+    // Guard: ref must share the same direct parent as task. In OmniJS the
+    // parent of a task is `task.parent` (the containing task or project).
+    // Compare by primaryKey so we never coerce across specifier types.
+    // Mirrors InMemoryAdapter.resolveReorderDestination — see #676.
+    const taskParentKey = task.parent?.id?.primaryKey ?? null;
+    const refParentKey = ref.parent?.id?.primaryKey ?? null;
+    if (taskParentKey !== refParentKey) {
+      return JSON.stringify({
+        error: {
+          code: "VALIDATION",
+          message: "reorderTask: reference task must share parent with the task being moved",
+        },
+      });
+    }
     location = mode === "before" ? ref.before : ref.after;
   } else if (mode === "start" || mode === "end") {
     let c;

